skin/src/main/java/org/researchstack/skin/task/SignUpTask.java

```diff
@@ -19,12 +36,82 @@
     public static final String ID_EMAIL    = "ID_EMAIL";
     public static final String ID_PASSWORD = "ID_PASSWORD";
 
+    private Map<String,Step> stepMap = new HashMap<>();
+    private Map<String,Boolean> answerMap = new HashMap<>();
+
     public SignUpTask(Context context)
     {
         super(TaskProvider.TASK_ID_SIGN_UP);
-        // creating here so it has access to context
-        inclusionCriteriaStep = UiManager.getInstance().getInclusionCriteriaStep(context);
-        inclusionCriteriaStep.setOptional(false);
+
+        initSteps(context);
+
+        inclusionCriteriaStep = stepMap.get(SignUpInclusionCriteriaStepIdentifier);
+
+    }
+
+  /**
+   * Create steps as defined in the JSON file.
+   *
+   * @param context
+   */
+  private void initSteps(Context context) {
+        InclusionCriteriaModel model = ResourceManager.getInstance()
+                .getInclusionCriteria()
+                .create(context);
+
+        for(InclusionCriteriaModel.Step s: model.steps) {
+            switch(s.type)
+            {
+                case "instruction":
```

Comment:
Gson will actually serialize enums automatically if you do @SerializedName above the enum.  This would be a way to avoid hard-coded case values for things like Step.type